### ISSUE TITLE
Guix container: `An unexpected error has occurred: AssertionError: BUG: expected environment for node to be healthy immediately after install`

### ISSUE THREAD

**Issue by alphapapa** [pre-commit/pre-commit#3085]
### search you tried in the issue tracker

`AssertionError`, `expected environment for node`, etc.

### describe your issue

Hello,

I'm trying to run `pre-commit` in a Guix container.  I've gotten pretty far by adding Guix packages to a manifest like so:
```scm
(specifications->manifest
  (list "pre-commit"
        "coreutils"
        "git"
        "nss-certs"
        "sed"
        "python-certifi"
        "le-certs"
        "python-pip"
        "fish"
        "less"
        "node"
        "which"))
```
Then I run this command in a repo which already uses `pre-commit`:
```
guix shell --container --network -m manifest.scm
```
Then inside of the container I run `pre-commit`, and it proceeds to configure several environments for the project, until it stops with this output:
```
[INFO] Installing environment for https://github.com/pre-commit/mirrors-prettier.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
An unexpected error has occurred: AssertionError: BUG: expected environment for node to be healthy immediately after install, please open an issue describing your environment

more info:

`node --version` returned 1
Check the log at /home/me/.cache/pre-commit/pre-commit.log
```
However, when I run `node --version` inside the container, it works and returns 0:
```
$ node --version; echo $?
v18.18.2
0
```
As well, FWIW, when I run `npm install prettier` in the container, it works without error.

I'm far from being an expert on Guix, and I'm new to `pre-commit`.  Please let me know what else I can do to help debug this.

Thanks for your work.

### pre-commit --version

3.3.3

### .pre-commit-config.yaml

```yaml
# Install pre-commit hooks via
# pre-commit install

repos:
  - repo: https://github.com/pre-commit/pre-commit-hooks
    rev: c4a0b883114b00d8d76b479c820ce7950211c99b # frozen: v4.5.0
    hooks:
      - id: check-added-large-files
      - id: check-ast
      - id: check-builtin-literals
      - id: check-case-conflict
      - id: check-json
      - id: check-toml
      - id: check-yaml
        args: [--allow-multiple-documents]
      - id: debug-statements
      - id: end-of-file-fixer
      - id: mixed-line-ending
      - id: trailing-whitespace

  - repo: https://github.com/pre-commit/mirrors-prettier
    rev: fc260393cc4ec09f8fc0a5ba4437f481c8b55dc1 # frozen: v3.0.3
    hooks:
      - id: prettier
        types_or: [yaml, toml, markdown, css, scss, javascript, json]
        args: [--prose-wrap=preserve]

  - repo: https://github.com/psf/black
    rev: 744d23b34800c06e10272149b70752396e90eeb8 # frozen: 23.10.1
    hooks:
      - id: black

  - repo: https://github.com/adamchainz/blacken-docs
    rev: 960ead214cd1184149d366c6d27ca6c369ce46b6 # frozen: 1.16.0
    hooks:
      - id: blacken-docs

  - repo: https://github.com/astral-sh/ruff-pre-commit
    rev: 3dacd83b1ee8fb950dbdd1001f9b5851728b6077 # frozen: v0.1.4
    hooks:
      - id: ruff
        args: [--fix, --exit-non-zero-on-fix, --target-version=py38]

ci:
  # This ensures that PR's aren't autofixed by the bot, rather you call
  # the bot to make the fix
  autofix_prs: false
  autofix_commit_msg: |
    '[pre-commit.ci <F0><9F><A4><96>] Apply code format tools to PR'
  # Update hook versions every month (so we don't get hit with weekly update PR's)
  autoupdate_schedule: monthly
```

### version information

```
pre-commit version: 3.3.3
git --version: git version 2.41.0
sys.version:
    3.10.7 (main, Jan  1 1970, 00:00:01) [GCC 11.3.0]
sys.executable: /gnu/store/78x787nrrr4y7rh7hfz37vbfh7hjm4z9-python-wrapper-3.10.7/bin/python
os.name: posix
sys.platform: linux
```

### error information

```
An unexpected error has occurred: AssertionError: BUG: expected environment for node to be healthy immed
iately after install, please open an issue describing your environment

more info:

`node --version` returned 1
```

```
Traceback (most recent call last):
  File "/gnu/store/c3v1vlsvpk6qnikaf3qm9sxnrv1a5221-pre-commit-3.3.3/lib/python3.10/site-packages/pre_co
mmit/error_handler.py", line 73, in error_handler
    yield
  File "/gnu/store/c3v1vlsvpk6qnikaf3qm9sxnrv1a5221-pre-commit-3.3.3/lib/python3.10/site-packages/pre_co
mmit/main.py", line 414, in main
    return run(args.config, store, args)
  File "/gnu/store/c3v1vlsvpk6qnikaf3qm9sxnrv1a5221-pre-commit-3.3.3/lib/python3.10/site-packages/pre_co
mmit/commands/run.py", line 442, in run
    install_hook_envs(to_install, store)
  File "/gnu/store/c3v1vlsvpk6qnikaf3qm9sxnrv1a5221-pre-commit-3.3.3/lib/python3.10/site-packages/pre_co
mmit/repository.py", line 248, in install_hook_envs
    _hook_install(hook)
  File "/gnu/store/c3v1vlsvpk6qnikaf3qm9sxnrv1a5221-pre-commit-3.3.3/lib/python3.10/site-packages/pre_co
mmit/repository.py", line 100, in _hook_install
    raise AssertionError(
AssertionError: BUG: expected environment for node to be healthy immediately after install, please open 
an issue describing your environment

more info:

`node --version` returned 1
```

**Comment by asottile** [comment]
you can reproduce via `nodeenv` without `pre-commit` -- guix and other nixlikes are known to not play well with `node` since they expect a particular layout / structure / etc. which is generally incompatible with a normal linux machine (which has things like `ld` and `libstdc++`)

likely `nodeenv nenv --prebuilt && . nenv/bin/activate && node --version`

**Comment by alphapapa** [comment]
Thanks.  For future reference to anyone else who might try to do this, here's where I am now and how I got there:

1.  Running `nodeenv nenv --prebuilt && . nenv/bin/activate && node --version` got to a Python SSL certificate error.  I resolved that with `export SSL_CERT_DIR=/etc/ssl/certs/`.
2. Running `nodeenv nenv --prebuilt && . nenv/bin/activate && node --version` again got to this error:
```
 * Install prebuilt node (21.4.0) ..... done.
node: error while loading shared libraries: libstdc++.so.6: cannot open shared object file: No such file or directory
```
I'm trying to figure out how to fix that in the Guix container, but I haven't found a solution yet.  Adding the `-F` argument to `guix container` was not enough, nor was adding `gcc-toolchain` to the manifest.  I'll probably have to ask some Guix folks next.

Thanks for your quick response.
